src/sage/modular/quasimodform/ring.py

```diff
@@ -778,3 +778,47 @@ def from_polynomial(self, polynomial):
             raise ValueError("the number of variables (%s) of the given polynomial cannot exceed the number of generators (%s) of the quasimodular forms ring" % (nb_var, self.ngens()))
         gens_dict = {poly_parent.gen(i):self.gen(i) for i in range(0, nb_var)}
         return self(polynomial.subs(gens_dict))
+
+    def basis_of_weight(self, weight):
+        r"""
+        Return a basis of elements generating the subspace of the given
+        weight.
+
+        INPUT:
+
+        - ``weight`` (integer) -- the weight of the subspace
+
+        OUTPUT:
+
+        A list of quasimodular forms of the given weight.
+
+        EXAMPLES::
+
+            sage: QM = QuasiModularForms(1)
+            sage: QM.basis_of_weight(12)
```

Comment:
It would be nice to check that this is indeed a basis, or at least that it generates elements. Perhaps generate a random element and express it as a linear combination of this basis? (Just use linear algebra with the first $20$ terms I guess.)

Reply:
I'm not sure I understand your question. The code (unless there's a typo) should give a basis since it is based on a mathematical fact.

Reply:
Nevermind then, it is not a requirement.